### PR TITLE
Show/hide card visualization for pinned collection items

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/23222

Adds a way to hide visualization for pinned questions in collections. Please note that the backend is not implemented yet and hiding visualization from the menu will do nothing.

How to test:
- Change https://github.com/metabase/metabase/blob/38e9a68b49a4102952de13328b707c057357c50d/frontend/src/metabase/collections/utils.ts#L85 to `item.description != null`
- Open a collection with questions of different visualization types. Questions with descriptions should be visualized, others should be listed at the bottom with static images.
- `Show/Don't show visualization` menu option should not be available for dashboards and non-pinned collection items

Screenshot 2022-06-14 at 17 06 43
Screenshot 2022-06-14 at 17 07 15
Screenshot 2022-06-14 at 17 07 20
Screenshot 2022-06-14 at 19 16 32
Screenshot 2022-06-14 at 19 16 38

### CODE REVIEW

File: frontend/src/metabase/collections/components/PinnedItemOverview/PinnedItemOverview.tsx [hunk 1]
```diff
@@ -54,39 +59,42 @@ function PinnedItemOverview({
...
-                  <PinnedChartCard
-                    item={item}
-                    metadata={metadata}
-                    collection={collection}
-                    bookmarks={bookmarks}
-                    onCopy={onCopy}
-                    onMove={onMove}
-                    onCreateBookmark={createBookmark}
-                    onDeleteBookmark={deleteBookmark}
+      {cardGroups.map(
```
Comment (ranquild) [inline frontend/src/metabase/collections/components/PinnedItemOverview/PinnedItemOverview.tsx:62]: Only `cardGroups` were added, the loop body is not changed

File: frontend/src/metabase/collections/components/PinnedQuestionCard/PinnedQuestionCard.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  collection: Collection;
+  metadata: Metadata;
+  bookmarks?: Bookmark[];
+  onCopy: (items: Item[]) => void;
+  onMove: (items: Item[]) => void;
+  onCreateBookmark?: (id: string, model: string) => void;
+  onDeleteBookmark?: (id: string, model: string) => void;
+}
+
+const PinnedQuestionCard = ({
```
Comment (ranquild) [inline frontend/src/metabase/collections/components/PinnedQuestionCard/PinnedQuestionCard.tsx:26]: I renamed this from `PinnedChartCard` to `PinnedQuestionCard`. Git tells it's a new file but it isn't

File: frontend/src/metabase/collections/components/PinnedQuestionCard/PinnedQuestionCard.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      <CardActionMenu
+        item={item}
+        collection={collection}
+        bookmarks={bookmarks}
+        onCopy={onCopy}
+        onMove={onMove}
+        createBookmark={onCreateBookmark}
+        deleteBookmark={onDeleteBookmark}
+      />
+      {isPreview ? (
```
Comment (ranquild) [inline frontend/src/metabase/collections/components/PinnedQuestionCard/PinnedQuestionCard.tsx:49]: The only change here apart from renaming is `isPreview` flag

File: frontend/src/metabase/collections/utils.ts [hunk 4]
```diff
@@ -4,17 +4,19 @@ import _ from "underscore";
 import { Collection, CollectionId } from "metabase-types/api";
 
 export type Item = {
+  id: number;
```
Comment (ranquild) [inline frontend/src/metabase/collections/utils.ts:7]: Fields, than methods

File: frontend/src/metabase/collections/utils.ts [hunk 5]
```diff
@@ -79,6 +81,10 @@ export function isItemPinned(item: Item) {
   return item.collection_position != null;
 }
 
+export function isPreviewEnabled(item: Item) {
+  return item.collection_preview ?? true;
```
Comment (ranquild) [inline frontend/src/metabase/collections/utils.ts:85]: Until the backend sets this field we need to default to `true`

File: frontend/src/metabase/visualizations/components/skeletons/AreaSkeleton/AreaSkeleton.tsx [hunk 6]
```diff
@@ -3,18 +3,18 @@ import SkeletonCaption from "../SkeletonCaption";
 import { SkeletonImage, SkeletonRoot } from "./AreaSkeleton.styled";
 
 export interface AreaSkeletonProps extends HTMLAttributes<HTMLDivElement> {
-  displayName?: string | null;
+  name?: string | null;
```
Comment (ranquild) [inline frontend/src/metabase/visualizations/components/skeletons/AreaSkeleton/AreaSkeleton.tsx:6]: Renamed `displayName` to `name` to match the backend API

**Review by alxnddr — APPROVED** [review]
Codewise looks good

**Pull request merged**